gammapy/image/profile.py

```diff
@@ -194,6 +196,241 @@ def save(self, filename):
+
+            gauss = Gaussian1DKernel(width)
+            err_sum = convolve(profile_err ** 2, gauss.array ** 2)
+            smoothed_err = np.sqrt(err_sum)
+        else:
+            raise ValueError("Not valid kernel choose either 'box' or 'gauss'")
+
+        table['profile'] = smoothed * self.table['profile'].unit
+        table['profile_err'] = smoothed_err * self.table['profile'].unit
+        return self.__class__(table)
+
+    def plot(self, ax=None, **kwargs):
+        """
+        Plot image profile.
+
+        Parameters
+        ----------
+        ax : `~matplotlib.axes.Axes`
+            Axes object
+        **kwargs : dict
+            Keyword arguments passed to plt.plot()
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`
+            Axes object
+        """
+        import matplotlib.pyplot as plt
+        
+        if ax is None:
+            ax = plt.gca()
+
+        y = self.table['profile'].data
+        x = self.x_ref.value
+        ax.plot(x, y, **kwargs)
+        ax.set_xlabel('lon')
+        ax.set_ylabel('profile')
+        ax.set_xlim(x.max(), x.min())
+        return ax
+
+    def plot_err(self, ax=None, **kwargs):
+        """
+        Plot image profile error as band.
+
+        Parameters
+        ----------
+        ax : `~matplotlib.axes.Axes`
+            Axes object
+        **kwargs : dict
+            Keyword arguments passed to plt.fill_between()
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`
+            Axes object
+        """
+        import matplotlib.pyplot as plt
+        
+        if ax is None:
+            ax = plt.gca()
+        
+        y = self.table['profile'].data
+        ymin = y - self.table['profile_err'].data
+        ymax = y + self.table['profile_err'].data
+        x = self.x_ref.value
+
+        # plotting defaults
+        kwargs.setdefault('alpha', 0.5)
+
+        ax.fill_between(x, ymin, ymax, **kwargs)
+        ax.set_xlabel('x (deg)')
+        ax.set_ylabel('profile')
+        return ax
+
+    @property
+    def x_ref(self):
+        """
+        Reference x coordinates.
+        """
+        return self.table['x_ref'].quantity
+
+    @property
+    def x_min(self):
+        """
+        Min. x coordinates.
+        """
+        return self.table['x_min'].quantity
+
+    @property
+    def x_max(self):
+        """
+        Max. x coordinates.
+        """
+        return self.table['x_max'].quantity
+
+    @property
+    def profile(self):
+        """
+        Image profile quantity.
+        """
+        return self.table['profile'].quantity
+
+    @property
+    def profile_err(self):
+        """
+        Image profile error quantity.
+        """
+        return self.table['profile'].quantity
+
+    def peek(self, figsize=(8, 4.5), **kwargs):
+        """
+        Show image profile and error.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Keyword arguments passed to plt.plot()
```

Comment:
Link to `plot` function in MPL, so that people can quickly look up the valid parameters?